packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -524,73 +486,44 @@ describe('TokenDetectionController', () => {
-        getNetworkClientById: jest.fn(),
-      },
-      {
-        disabled: false,
-        isDetectionEnabledForNetwork: true,
-        isDetectionEnabledFromPreferences: false,
-        selectedAddress: '0x1',
-      },
-    );
+    tokenDetection = new TokenDetectionController({
+      chainId: ChainId.mainnet,
+      selectedAddress: '0x1',
+      onPreferencesStateChange,
+      getBalancesInSingleCall: getBalancesInSingleCallMock,
+      addDetectedTokens: stub,
+      getTokensState: () => tokensController.state,
+      getPreferencesState: () => preferences.state,
+      messenger: buildTokenDetectionControllerMessenger(controllerMessenger),
+    });
 
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     await preferencesStateChangeListener!({
       selectedAddress: '0x1',
       useTokenDetection: true,
     });
-
+    await tokenDetection.start();
     expect(getBalancesInSingleCallMock.called).toBe(true);
   });
 
-  it('should call getBalancesInSingleCall if onNetworkStateChange is called with a chainId that supports token detection and is changed', async () => {
+  it('should call getBalancesInSingleCall if network is changed to a chainId that supports token detection', async () => {
     const stub = sinon.stub();
     const getBalancesInSingleCallMock = sinon.stub();
-    let networkStateChangeListener: (state: any) => void;
-    const onNetworkStateChange = sinon.stub().callsFake((listener) => {
-      networkStateChangeListener = listener;
-    });
-    tokenDetection = new TokenDetectionController(
-      {
-        onNetworkStateChange,
-        onTokenListStateChange: stub,
-        onPreferencesStateChange: stub,
-        getBalancesInSingleCall: getBalancesInSingleCallMock,
-        addDetectedTokens: stub,
-        getTokensState: () => tokensController.state,
-        getTokenListState: () => tokenList.state,
-        getNetworkState: () => defaultNetworkState,
-        getPreferencesState: () => preferences.state,
-        getNetworkClientById: jest.fn(),
-      },
-      {
-        disabled: false,
-        isDetectionEnabledFromPreferences: true,
-        chainId: SupportedTokenDetectionNetworks.polygon,
-        isDetectionEnabledForNetwork: true,
-        selectedAddress: '0x1',
-      },
-    );
-
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    await networkStateChangeListener!({
-      providerConfig: { chainId: ChainId.mainnet },
+    tokenDetection = new TokenDetectionController({
+      chainId: SupportedTokenDetectionNetworks.polygon,
+      selectedAddress: '0x1',
+      onPreferencesStateChange: stub,
+      getBalancesInSingleCall: getBalancesInSingleCallMock,
+      addDetectedTokens: stub,
+      getTokensState: () => tokensController.state,
+      getPreferencesState: () => preferences.state,
+      messenger: buildTokenDetectionControllerMessenger(controllerMessenger),
     });
 
+    changeNetwork(mainnet);
+    await tokenDetection.start();
```

Comment:
Why is this necessary now?

Reply:
Same as above: https://github.com/MetaMask/core/pull/3609#discussion_r1423396494

Reply:
Restored here https://github.com/MetaMask/core/pull/3609/commits/3532bbc275187e36dcd2aea5cd2aa98479fd3ab6 based on our discussion on removing `deferPollingStart`.